pkg/agent/manager/cache/cache.go

```diff
@@ -59,25 +63,23 @@ func New(log logrus.FieldLogger, bundle []*x509.Certificate) *cacheImpl {
 	return &cacheImpl{
 		cache:       make(map[string]*Entry),
 		log:         log.WithField("subsystem_name", "cache"),
-		bundle:      bundle,
+		bundle:      observer.NewProperty(bundle),
 		subscribers: NewSubscribers(),
 	}
 }
 
 func (c *cacheImpl) SetBundle(bundle []*x509.Certificate) {
-	c.m.Lock()
-	c.bundle = bundle
-	c.m.Unlock()
-
+	c.bundle.Update(bundle)
 	subs := c.subscribers.getAll()
 	c.notifySubscribers(subs)
 }
 
 func (c *cacheImpl) Bundle() (result []*x509.Certificate) {
-	c.m.Lock()
-	defer c.m.Unlock()
-	result = append(result, c.bundle...)
-	return result
+	return append(result, c.bundle.Value().([]*x509.Certificate)...)
```

Comment:
if we drop the named return value, could this be simplified to `return c.bundle.Value().([]*x509.Certificate)`?